components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2925,4 +2925,21 @@ private boolean isAccessTokenClaimsSeparationEnabledForApp(String consumerKey, S
         return OAuth2Util.isAppVersionAllowed(serviceProvider.getApplicationVersion(),
                 ApplicationConstants.ApplicationVersion.APP_VERSION_V2);
     }
+
+    private static String getAppTenantDomain() throws IdentityOAuthAdminException {
+
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String applicationResidentOrgId = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getApplicationResidentOrganizationId();
+        if (StringUtils.isNotEmpty(applicationResidentOrgId)) {
+            try {
+                tenantDomain = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
```

Comment:
why we need this change? can 
```
CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
``` and 
```
resolveTenantDomain(applicationResidentOrgId);
``` be different? if so in what context>

Reply:
Here what we are doing is that, if the `applicationResidentOrgId` is available in the context then, we are creating the application in that specific organization. This param will be set from the rest valve[1] and in the priviledged carbon context, that param is stored[2]

[1] https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/298
[2] https://github.com/wso2/carbon-kernel/pull/4104